extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlRolloutHistory.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    public History(long revision, String changeCause) {
+      this.revision = revision;
+      this.changeCause = changeCause;
+    }
+
+    @Override
+    public String toString() {
+      return "{revision :" + revision + ",changeCause :" + changeCause + "}";
+    }
+  }
+
+  private long revision;
+
+  private final List<History> histories;
+
+  private V1PodTemplateSpec template;
+
+  @Override
+  public ApiType execute() throws KubectlException {
+    validate();
+    AppsV1Api api = new AppsV1Api(this.apiClient);
+    try {
+      if (apiTypeClass.equals(V1Deployment.class)) {
+        V1Deployment deployment = api.readNamespacedDeployment(name, namespace, null, null, null);
+        deploymentViewHistory(deployment, api);
+        return (ApiType) deployment;
+      } else {
+        throw new KubectlException("Unsupported class for rollout history: " + apiTypeClass);
+      }
+    } catch (ApiException ex) {
+      throw new KubectlException(ex);
+    }
+  }
+
+  KubectlRolloutHistory(Class<ApiType> apiTypeClass) {
+    super(apiTypeClass);
+    revision = 0;
+    histories = new ArrayList<>();
+  }
+
+  public KubectlRolloutHistory<ApiType> revision(int revision) {
+    this.revision = revision;
+    return this;
+  }
+
+  private void validate() throws KubectlException {
+    StringBuilder msg = new StringBuilder();
+    if (name == null) {
+      msg.append("Missing name, ");
+    }
+    if (namespace == null) {
+      msg.append("Missing namespace, ");
+    }
+    if (revision < 0) {
+      msg.append("revision must be a positive integer: ").append(revision);
+    }
+    if (msg.length() > 0) {
+      throw new KubectlException(msg.toString());
+    }
+  }
+
+  private void deploymentViewHistory(V1Deployment deployment, AppsV1Api api) throws ApiException {
+    List<V1ReplicaSet> allOldRSs = new ArrayList<>();
+    List<V1ReplicaSet> oldRSs = new ArrayList<>();
+    V1ReplicaSet newRs = DeploymentHelper.getAllReplicaSets(deployment, api, oldRSs, allOldRSs);
+    if (newRs != null) {
+      allOldRSs.add(newRs);
+    }
+    Map<Long, V1PodTemplateSpec> historyInfo = new HashMap<>();
+    for (V1ReplicaSet rs : allOldRSs) {
+      Long v = DeploymentHelper.revision(rs.getMetadata());
+      historyInfo.put(v, rs.getSpec().getTemplate());
+      String changeCause = getChangeCause(rs.getMetadata());
+      if (historyInfo.get(v).getMetadata().getAnnotations() == null) {
+        historyInfo.get(v).getMetadata().setAnnotations(new HashMap<>());
+      }
+      if (changeCause != null && changeCause.length() > 0) {
+        historyInfo.get(v).getMetadata().getAnnotations().put(CHANGE_CAUSE_ANNOTATION, changeCause);
+      }
+    }
+
+    if (revision > 0) {
+      // get details of a specific revision
+      V1PodTemplateSpec template = historyInfo.get(revision);
+      if (template == null) {
+        throw new ApiException(
+            HttpsURLConnection.HTTP_NOT_FOUND, "unable to find the specified revision " + revision);
```

Comment:
I think that this is going to confuse people, because there's no actual web 404 involved here, just pass the string w/o the 404